scheduler/src/cook/mesos/util.clj

```diff
@@ -198,6 +198,39 @@
      get-pending-jobs-duration
      (get-pending-job-ents* unfiltered-db true))))
 
+(defn generate-intervals
+  "Generates a list of intervals between start and end as pairs
+   [interval-start interval-end].
+
+   Parameters:
+   ----------
+   start : clj-time/datetime
+   end : clj-time/datetime
+   interval->steps : fn
+      function that takes a clj-time/interval and returns the number of steps
+      based on the step-fn between start and end
+   step-fn : fn
+      function that returns a clj-time/timedelta given n steps
+
+   Returns:
+   --------
+   list of pairs, [interval-start interval-end]"
+  ([start end]
+   (generate-intervals start end t/in-days t/days))
+  ([start end interval->steps step-fn]
+   (if (t/before? end start)
+     []
+     (->> (for [steps-from-start (range 0 (inc (interval->steps (t/interval start end))))]
+            (let [interval-start (t/plus start (step-fn steps-from-start))
+                  interval-end (t/plus start (step-fn (inc steps-from-start)))
+                  interval-end (if (t/before? interval-end end)
+                                 interval-end
+                                 end)]
+              [interval-start interval-end]))
+          ;; This filter is here because joda time t/days returns the floor of days if there is an
+          ;; a non integer number of days. 
+          (filter (fn [[s e]] (> (t/in-millis (t/interval s e)) 0)))))))
+
 (defn get-jobs-by-user-and-state
```

Comment:
What about passing an optional limit here to stop iterating once you've hit the limit?